pxf/pxf-json/src/main/java/org/apache/hawq/pxf/plugins/json/JsonInputFormat.java

```diff
@@ -0,0 +1,226 @@
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.security.InvalidParameterException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapred.FileInputFormat;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.LineRecordReader;
+import org.apache.hadoop.mapred.RecordReader;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.log4j.Logger;
+import org.codehaus.jackson.JsonFactory;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParseException;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+
+public class JsonInputFormat extends FileInputFormat<Text, NullWritable> {
+
+	private static JsonFactory factory = new JsonFactory();
+	private static ObjectMapper mapper = new ObjectMapper(factory);
+
+	public static final String ONE_RECORD_PER_LINE = "json.input.format.one.record.per.line";
+	public static final String RECORD_IDENTIFIER = "json.input.format.record.identifier";
+
+	@Override
+	public RecordReader<Text, NullWritable> getRecordReader(InputSplit split, JobConf conf, Reporter reporter)
+			throws IOException {
+
+		if (conf.getBoolean(ONE_RECORD_PER_LINE, false)) {
+
+			return new SimpleJsonRecordReader(conf, (FileSplit) split);
+		} else {
+			return new JsonRecordReader(conf, (FileSplit) split);
+		}
+	}
+
+	public static class SimpleJsonRecordReader implements RecordReader<Text, NullWritable> {
+
+		private LineRecordReader rdr = null;
+		private LongWritable key = new LongWritable();
+		private Text value = new Text();
+
+		public SimpleJsonRecordReader(Configuration conf, FileSplit split) throws IOException {
+			rdr = new LineRecordReader(conf, split);
+		}
+
+		@Override
+		public void close() throws IOException {
+			rdr.close();
+		}
+
+		@Override
+		public Text createKey() {
+			return value;
+		}
+
+		@Override
+		public NullWritable createValue() {
+			return NullWritable.get();
+		}
+
+		@Override
+		public long getPos() throws IOException {
+			return rdr.getPos();
+		}
+
+		@Override
+		public boolean next(Text key, NullWritable value) throws IOException {
```

Comment:
As those methods implement `RecordReader` i guess inherit the the doc (`{@inheritDoc}`) should be enough as documentation? Only `next()` seems to deserve additional doc.